redgenes/bakta_annotations.py

```diff
@@ -0,0 +1,189 @@
+from sql_connection import TRN
+import pandas as pd
+from pathlib import Path
+from collections import defaultdict
+from sql_initialize_db import initialize_db
+from utils import (
+    copy_and_unzip,
+    run_command_and_check_outputs,
+    read_gff_file,
+    extract_gff_info,
+    process_gff_info,
+)
+from exceptions import (
+    InvalidInputTsv,
+    BaktaError,
+)
+from sql_connection import TRN
+
+bakta_version = 1
+bakta_commands = 1
+
+# Ensure all necessary imports are included
+def process_dbxref(dbxref):
+    """
+    Process dbxref entry and return a structured dictionary.
+    """
+    entry_dict = defaultdict(list)
+    if pd.notna(dbxref):
+        entries = dbxref.split(';')
+        for entry in entries:
+            if ':' in entry:
+                entry_type, entry_value = entry.split(':', 1)
+                entry_dict[entry_type.strip()].append(entry_value.strip())
+    return entry_dict
+
+def insert_dbxref_info(bakta_accesion, dbxref_data):
+    """
+    Insert dbxref information into corresponding tables.
+    """
+    for dbxref_type, accession_list in dbxref_data.items():
+        table_name = dbxref_type.lower()  # Assuming table names are lowercase versions of dbxref types
+        if table_name in ['kegg', 'refseq', 'uniparc', 'uniref', 'so', 'pfam']:  # Check if table_name is valid
+            sql = f"INSERT INTO {table_name} (bakta_accession, {table_name.upper()}) VALUES (?, ?)"
+            for accession in accession_list:
+                TRN.add(sql, [bakta_accesion, accession])
```

Comment:
List lookups are O(N), whereas set lookups are O(1)

```suggestion
    valid_tables = {'kegg', 'refseq', 'uniparc', 'uniref', 'so', 'pfam'} 
    for dbxref_type, accession_list in dbxref_data.items():
        table_name = dbxref_type.lower()  # Assuming table names are lowercase versions of dbxref types
        if table_name in valid_tables:
            sql = f"INSERT INTO {table_name} (bakta_accession, {table_name.upper()}) VALUES (?, ?)"
            for accession in accession_list:
                TRN.add(sql, [bakta_accesion, accession])
```